### ISSUE TITLE
bindfs not working - Error: Operation not permitted

### ISSUE THREAD

**Issue by MPParsley** [puphpet/puphpet#1493]
Anyone able to see what is going wrong?

==> default: Info: /Stage[main]/Main/File[/var/www/html/index.html]: Filebucketed /var/www/html/index.html to puppet with sum 74cec59a19e5d16f7cc6a2445e35fa3b
==> default: Error: Operation not permitted - /var/www/html/index.html20150401-9719-q6h7bh
==> default: Error: /Stage[main]/Main/File[/var/www/html/index.html]/content: change from {md5}74cec59a19e5d16f7cc6a2445e35fa3b to {md5}85a571ae4985a6c6b49dadaac8f3a54d failed: Operation not permitted - /var/www/html/index.html20150401-9719-q6h7bh
==> default: Notice: /Stage[main]/Main/File[/var/www/html/index.html]/owner: owner changed 'www-data' to 'root'
==> default: Info: Creating state file /var/lib/puppet/state/state.yaml
==> default: Notice: Finished catalog run in 705.46 seconds
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.

```vagrantfile-local:
    vm:
        box: puphpet/debian75-x64
        box_url: puphpet/debian75-x64
        hostname: ''
        memory: '4096'
        cpus: '4'
        chosen_provider: virtualbox
        network:
            private_network: 192.168.55.101
            forwarded_port:
                Cgq7pqhZ6PZW:
                    host: '5522'
                    guest: '22'
                vXmyIih6EjNk:
                    host: '5580'
                    guest: '56'
                cYlP2dAOVNPV:
                    host: '5506'
                    guest: '3306'
                Gmu92NqENGH5:
                    host: '5581'
                    guest: '1080'
                ChfUKQZQ8TB4:
                    host: '5583'
                    guest: '8983'
        post_up_message: ''
        provider:
            virtualbox:
                modifyvm:
                    natdnshostresolver1: on
            vmware:
                numvcpus: 1
            parallels:
                cpus: 1
        provision:
            puppet:
                manifests_path: puphpet/puppet
                manifest_file: site.pp
                module_path: puphpet/puppet/modules
                options:
                    - '--verbose'
                    - '--hiera_config /vagrant/puphpet/puppet/hiera.yaml'
                    - '--parser future'
        synced_folder:
            IAJPhBYCEkv6:
                owner: www-data
                group: www-data
                source: /Volumes/webdev/www
                target: /var/www
                sync_type: default
                rsync:
                    args:
                        - '--verbose'
                        - '--archive'
                        - '-z'
                    exclude:
                        - .vagrant/
                    auto: 'false'
        usable_port_range:
            start: 10200
            stop: 10500
    ssh:
        host: null
        port: null
        private_key_path: null
        username: vagrant
        guest_port: null
        keep_alive: true
        forward_agent: false
        forward_x11: false
        shell: 'bash -l'
    vagrant:
        host: detect
server:
    install: '1'
    packages:
        - htop
users_groups:
    install: '1'
    groups:
        - staff
    users:
        - c4m
        - 'maartensegers:staff'
cron:
    install: '1'
    jobs: {  }
firewall:
    install: '1'
    rules: {  }
apache:
    install: '1'
    settings:
        user: www-data
        group: www-data
        default_vhost: true
        manage_user: false
        manage_group: false
        sendfile: 0
    modules:
        - deflate
        - expires
        - headers
        - proxy_fcgi
        - rewrite
    vhosts:
        l1sjifzqubo8:
            servername: capacity4more.eu.dev
            serveraliases:
                - capacity4more.eu.vm
            docroot: /var/www/c4m/www
            port: '80'
            setenv:
                - 'APP_ENV dev'
                - 'APPLICATION_ENV development'
            custom_fragment: ''
            directories:
                y24axfac1c4i:
                    provider: directory
                    path: /var/www/c4m/www
                    options:
                        - Indexes
                        - FollowSymlinks
                        - MultiViews
                    allow_override:
                        - All
                    require:
                        - 'all granted'
                    custom_fragment: ''
                    files_match:
                        xr3tnt3pb2i8:
                            provider: filesmatch
                            path: \.php$
                            sethandler: 'proxy:fcgi://127.0.0.1:9000'
                            custom_fragment: ''
            ssl_cert: ''
            ssl_key: ''
            ssl_chain: ''
            ssl_certs_dir: ''
nginx:
    install: '0'
    settings:
        default_vhost: 1
        proxy_buffer_size: 128k
        proxy_buffers: '4 256k'
    upstreams: {  }
    vhosts:
        eohx2o7dx1q1:
            proxy: ''
            server_name: awesome.dev
            server_aliases:
                - www.awesome.dev
            www_root: /var/www/awesome
            listen_port: '80'
            index_files:
                - index.html
                - index.htm
                - index.php
            client_max_body_size: 1m
            locations:
                e8aeixm7czcg:
                    location: /
                    try_files:
                        - $uri
                        - $uri/
                        - index.php
                    fastcgi: ''
                    fastcgi_index: ''
                    fastcgi_split_path: ''
                s7h46mbzidux:
                    location: '~ \.php$'
                    try_files:
                        - $uri
                        - $uri/
                        - index.php
                        - /index.php$is_args$args
                    fastcgi: '127.0.0.1:9000'
                    fastcgi_index: index.php
                    fastcgi_split_path: '^(.+\.php)(/.*)$'
                    fast_cgi_params_extra:
                        - 'SCRIPT_FILENAME $request_filename'
                        - 'APP_ENV dev'
            ssl_cert: ''
            ssl_key: ''
            ssl_port: '443'
php:
    install: '1'
    settings:
        version: '56'
    fpm_settings:
        port: 9000
    composer: '1'
    composer_home: ''
    modules:
        php:
            - cli
            - intl
            - mcrypt
            - gd
            - mysql
            - curl
        pear: {  }
        pecl:
            - pecl_http
    ini:
        display_errors: On
        error_reporting: '-1'
        session.save_path: /var/lib/php/session
        date.timezone: UTC
        upload_max_filesize: 32M
        post_max_size: 64M
        memory_limit: '-1'
hhvm:
    install: '0'
    nightly: 0
    composer: '1'
    composer_home: ''
    settings:
        host: 127.0.0.1
        port: '9000'
    ini:
        display_errors: On
        error_reporting: '-1'
    timezone: null
xdebug:
    install: '1'
    settings:
        xdebug.default_enable: '1'
        xdebug.remote_autostart: '0'
        xdebug.remote_connect_back: '1'
        xdebug.remote_enable: '1'
        xdebug.remote_handler: dbgp
        xdebug.remote_port: '9000'
xhprof:
    install: '0'
wpcli:
    install: '0'
    version: v0.17.1
drush:
    install: '1'
    version: 6.3.0
ruby:
    install: '1'
    versions:
        bTMtyWB04xu5:
            default: '0'
            version: ''
            bundler: '1'
nodejs:
    install: '1'
    npm_packages:
        - grunt-cli
        - bower
python:
    install: '1'
    packages:
        - meld3
    versions: {  }
mysql:
    install: '1'
    settings:
        version: '5.6'
        root_password: drupal
        override_options: {  }
    adminer: '1'
    users:
        Y5VjHcEhEwcA:
            name: amp
            password: drupal
    databases:
        ZCKzNH1d78bh:
            name: amp_c4m
            sql: ''
        wK0sLOvEqhG1:
            name: amp_c4d
            sql: ''
    grants:
        eWd4H2GfMxKC:
            user: amp
            table: '*.*'
            privileges:
                - ALL
postgresql:
    install: '0'
    settings:
        global:
            encoding: UTF8
            version: '9.3'
        server:
            postgres_password: '123'
    databases: {  }
    users: {  }
    grants: {  }
    adminer: 0
sqlite:
    install: '0'
    adminer: 0
    databases: {  }
mongodb:
    install: '0'
    settings:
        auth: 1
        bind_ip: 127.0.0.1
        port: '27017'
    databases: {  }
redis:
    install: '0'
    settings:
        conf_port: '6379'
mailcatcher:
    install: '1'
    settings:
        smtp_ip: 0.0.0.0
        smtp_port: 1025
        http_ip: 0.0.0.0
        http_port: '1080'
        mailcatcher_path: /usr/local/rvm/wrappers/default
        from_email_method: inline
beanstalkd:
    install: '0'
    settings:
        listenaddress: 0.0.0.0
        listenport: '13000'
        maxjobsize: '65535'
        maxconnections: '1024'
        binlogdir: /var/lib/beanstalkd/binlog
        binlogfsync: null
        binlogsize: '10485760'
    beanstalk_console: 0
    binlogdir: /var/lib/beanstalkd/binlog
rabbitmq:
    install: '0'
    settings:
        port: '5672'
    users: {  }
    vhosts: {  }
    plugins: {  }
elastic_search:
    install: '0'
    settings:
        version: 1.4.1
        java_install: true
solr:
    install: '1'
    settings:
        version: 4.10.2
        port: '8983'
```

**Comment by adrenth-exed** [comment]
I got the same issue here:

==> default: Error: Operation not permitted - /var/www/html/index.html20150402-21322-f0sm7s
==> default: Error: /Stage[main]/Main/File[/var/www/html/index.html]/content: change from {md5}74cec59a19e5d16f7cc6a2445e35fa3b to {md5}85a571ae4985a6c6b49dadaac8f3a54d failed: Operation not permitted - /var/www/html/index.html20150402-21322-f0sm7s

Some more details regarding this issue:

OS X 10.10.2

in puppet/nodes/Apache.pp:222

When the puppet:///modules/puphpet/webserver_landing.erb file is copied to /var/www/html/index.html, this index.html file gets permissions 644 instead of 664.

-rw-r--r--   1 alwin  staff  11104 Apr  2 16:44 index.html

To get it to work is to remove /var/www/html and provision again. 

A fragment from my config.yml:

```yaml
apache:
    install: '1'
    settings:
        user: www-data
        group: www-data
        default_vhost: true
        manage_user: false
        manage_group: false
        sendfile: 0
    modules:
        - proxy_fcgi
        - headers
        - rewrite
        - expires
        - vhost_alias
    vhosts:
        exed_dev_443:
            servername: exed.dev
            serveraliases:
                - '*.exed.dev'
            docroot: /var/www/sites
            port: '443'
            custom_fragment: '/var/www/puphpet/exed/apache/custom_fragment_VirtualHost.txt'
            directories:
                bgl06ezgv2mw:
                    provider: directory
                    path: /var/www/sites
                    options:
                        - Indexes
                        - FollowSymlinks
                        - MultiViews
                    allow_override:
                        - All
                    require:
                        - 'all granted'
                    custom_fragment: '/var/www/puphpet/exed/apache/custom_fragment_Directory.txt'
                    files_match:
                        1sj56kfu4opx:
                            provider: filesmatch
                            path: \.php$
                            sethandler: 'proxy:fcgi://127.0.0.1:9000'
                            custom_fragment: '/var/www/puphpet/exed/apache/custom_fragment_FilesMatch.txt'
            ssl: '1'
            ssl_cert: ''
            ssl_key: ''
            ssl_chain: ''
            ssl_certs_dir: ''
```

**Comment by MPParsley** [comment]
Tried a number of things, including the "remove /var/www/html" tip above, alas none of them seem to work. Tried several times to uninstall & install vagrant bindfs.
The only way I could get it to work was by by setting sync_type back to "default" instead of "nfs".

Seems like there's still a bug wrt nfs.

**Comment by vancouverwill** [comment]
I switched 'sync_type' to 'default' and then that removed the error for me too.

ox v10.10.2
Vagrant 1.7.2
virtualbox v4.3.26

**Comment by MPParsley** [comment]
Unfortunately this doesn't come with the best performance.

**Comment by jtreminio** [comment]
Fixed by a106ebb
